src/eventsBuffer.ts

```diff
@@ -137,6 +137,19 @@ export class EventsBuffer<T> {
 		this.maybeDropOlderThen(time - this.maxAge);
 	}
 
+	/**
+	 * Wait for all events to be processed by the pusher.
+	 */
+	async drain(): Promise<void> {
+		if (!this.callback) {
+			return;
+		}
+
+		while (this.pusherRunning || this.events.length > 0) {
```

Comment:
This while loop is not great, it's busy loop, and actually doesn't allow to execute any code beyond it.

I don't know how familiar you are with Node.js, but this isn't a single queue where all has the same priority, there is a hierarchy of queues and by fully saturating `nextTick` queue and microtask queue (promises) there is no room for any timers and io and so on to get scheduled.

Here is article about it https://nodejs.org/en/learn/asynchronous-work/event-loop-timers-and-nexttick